internal/provider/kubernetes/controller.go

```diff
@@ -1181,6 +1219,28 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 		return err
 	}
 
+	// Watch Backend CRUDs and process affected *Route objects.
+	if r.envoyGateway.ExtensionAPIs != nil && r.envoyGateway.ExtensionAPIs.EnableBackend {
+		backendPredicates := []predicate.TypedPredicate[*egv1a1.Backend]{
+			predicate.NewTypedPredicateFuncs[*egv1a1.Backend](func(be *egv1a1.Backend) bool {
```

Comment:
ObservedGeneration predicate ? we haven't added the `ObjectMeta` yet so it won't work
the issue I see here is constant reconciling - 
* reconcile backend
* translate backend and update status
* translate again (may be stopped by watchable layer which checks for equality)

Reply:
https://github.com/envoyproxy/gateway/blob/44330ef0ea6b0d6725e2cb40ab7b7837dccd1fb7/internal/gatewayapi/resource.go#L172

Reply:
Can you elaborate on why `TypedGenerationChangedPredicate` won't work here?

Reply:
nvm it will, took a look at the struct https://github.com/envoyproxy/gateway/blob/6f30397e23ed06218896b7e5a0a8dad3368775f3/api/v1alpha1/backend_types.go#L43
and you've added `ObjectMeta`,  so should work, lets add it